common/persistence/src/main/java/app/coronawarn/datadonation/common/persistence/service/SaltService.java

```diff
@@ -22,8 +23,8 @@
   public void deleteSalt(String salt) {
     try {
       saltRepository.deleteSalt(salt);
-    } catch (RuntimeException saltNotFoundException) {
-      logger.error("Salt " + salt + " not found.");
+    } catch (RuntimeException ex) {
+      throw new DeleteSaltException(salt);
```

Comment:
still... what if `deleteSalt` throws another RuntimeException?
now you are throwing a new exception, but still silently forget about the root cause...

> Direct Known Subclasses:
[AnnotationTypeMismatchException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/annotation/AnnotationTypeMismatchException.html), [ArithmeticException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ArithmeticException.html), [ArrayStoreException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ArrayStoreException.html), [BufferOverflowException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/BufferOverflowException.html), [BufferUnderflowException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/BufferUnderflowException.html), [CannotRedoException](https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/swing/undo/CannotRedoException.html), [CannotUndoException](https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/swing/undo/CannotUndoException.html), [CatalogException](https://docs.oracle.com/en/java/javase/11/docs/api/java.xml/javax/xml/catalog/CatalogException.html), [ClassCastException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ClassCastException.html), [ClassNotPreparedException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/ClassNotPreparedException.html), [CMMException](https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/java/awt/color/CMMException.html), [CompletionException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/CompletionException.html), [ConcurrentModificationException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ConcurrentModificationException.html), [DateTimeException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/DateTimeException.html), [DOMException](https://docs.oracle.com/en/java/javase/11/docs/api/java.xml/org/w3c/dom/DOMException.html), [DuplicateRequestException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/request/DuplicateRequestException.html), [EmptyStackException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/EmptyStackException.html), [EnumConstantNotPresentException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/EnumConstantNotPresentException.html), [EventException](https://docs.oracle.com/en/java/javase/11/docs/api/java.xml/org/w3c/dom/events/EventException.html), [FileSystemAlreadyExistsException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/FileSystemAlreadyExistsException.html), [FileSystemNotFoundException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/FileSystemNotFoundException.html), [FindException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/module/FindException.html), [IllegalArgumentException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/IllegalArgumentException.html), [IllegalCallerException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/IllegalCallerException.html), [IllegalMonitorStateException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/IllegalMonitorStateException.html), [IllegalPathStateException](https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/java/awt/geom/IllegalPathStateException.html), [IllegalStateException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/IllegalStateException.html), [IllformedLocaleException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/IllformedLocaleException.html), [ImagingOpException](https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/java/awt/image/ImagingOpException.html), [InaccessibleObjectException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/reflect/InaccessibleObjectException.html), [IncompleteAnnotationException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/annotation/IncompleteAnnotationException.html), [InconsistentDebugInfoException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/InconsistentDebugInfoException.html), [IndexOutOfBoundsException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/IndexOutOfBoundsException.html), [InternalException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/InternalException.html), [InvalidCodeIndexException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/InvalidCodeIndexException.html), [InvalidLineNumberException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/InvalidLineNumberException.html), [InvalidModuleDescriptorException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/module/InvalidModuleDescriptorException.html), [InvalidModuleException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/InvalidModuleException.html), [InvalidRequestStateException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/request/InvalidRequestStateException.html), [InvalidStackFrameException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/InvalidStackFrameException.html), [JarSignerException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jartool/jdk/security/jarsigner/JarSignerException.html), [JMRuntimeException](https://docs.oracle.com/en/java/javase/11/docs/api/java.management/javax/management/JMRuntimeException.html), [JSException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jsobject/netscape/javascript/JSException.html), [LayerInstantiationException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/LayerInstantiationException.html), [LSException](https://docs.oracle.com/en/java/javase/11/docs/api/java.xml/org/w3c/dom/ls/LSException.html), [MalformedParameterizedTypeException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/reflect/MalformedParameterizedTypeException.html), [MalformedParametersException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/reflect/MalformedParametersException.html), [MirroredTypesException](https://docs.oracle.com/en/java/javase/11/docs/api/java.compiler/javax/lang/model/type/MirroredTypesException.html), [MissingResourceException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/MissingResourceException.html), [NashornException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.scripting.nashorn/jdk/nashorn/api/scripting/NashornException.html), [NativeMethodException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/NativeMethodException.html), [NegativeArraySizeException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/NegativeArraySizeException.html), [NoSuchDynamicMethodException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.dynalink/jdk/dynalink/NoSuchDynamicMethodException.html), [NoSuchElementException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/NoSuchElementException.html), [NoSuchMechanismException](https://docs.oracle.com/en/java/javase/11/docs/api/java.xml.crypto/javax/xml/crypto/NoSuchMechanismException.html), [NullPointerException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/NullPointerException.html), [ObjectCollectedException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/ObjectCollectedException.html), [ProfileDataException](https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/java/awt/color/ProfileDataException.html), [ProviderException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/security/ProviderException.html), [ProviderNotFoundException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/ProviderNotFoundException.html), [RangeException](https://docs.oracle.com/en/java/javase/11/docs/api/java.xml/org/w3c/dom/ranges/RangeException.html), [RasterFormatException](https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/java/awt/image/RasterFormatException.html), [RejectedExecutionException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/RejectedExecutionException.html), [ResolutionException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/module/ResolutionException.html), [SecurityException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/SecurityException.html), [SPIResolutionException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jshell/jdk/jshell/spi/SPIResolutionException.html), [TypeNotPresentException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/TypeNotPresentException.html), [UncheckedIOException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/UncheckedIOException.html), [UndeclaredThrowableException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/reflect/UndeclaredThrowableException.html), [UnknownEntityException](https://docs.oracle.com/en/java/javase/11/docs/api/java.compiler/javax/lang/model/UnknownEntityException.html), [UnknownTreeException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.scripting.nashorn/jdk/nashorn/api/tree/UnknownTreeException.html), [UnmodifiableModuleException](https://docs.oracle.com/en/java/javase/11/docs/api/java.instrument/java/lang/instrument/UnmodifiableModuleException.html), [UnmodifiableSetException](https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/print/attribute/UnmodifiableSetException.html), [UnsupportedOperationException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/UnsupportedOperationException.html), [VMDisconnectedException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/VMDisconnectedException.html), [VMMismatchException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/VMMismatchException.html), [VMOutOfMemoryException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jdi/com/sun/jdi/VMOutOfMemoryException.html), [WrongMethodTypeException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/invoke/WrongMethodTypeException.html), [XPathException](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.xml.dom/org/w3c/dom/xpath/XPathException.html)